packages/hive/src/Adaptor.js

```diff
@@ -0,0 +1,154 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+
+import hive from 'hive-driver';
+
+const { TCLIService, TCLIService_types } = hive.thrift;
+const utils = new hive.HiveUtils(TCLIService_types);
+
+const logger = message => console.log(message);
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      connect,
+      ...operations,
+      cleanupState
+    )({
+      ...initialState,
+      ...state,
+    });
+  };
+}
+
+function connect(state) {
+  const { host, port, username, password } = state.configuration;
+
+  const client = new hive.HiveClient(TCLIService, TCLIService_types);
+
+  const connection = new hive.connections.TcpConnection();
+  const auth =
+    username && password
+      ? new hive.auth.PlainTcpAuthentication({
+          username,
+          password,
+        })
+      : new hive.auth.NoSaslAuthentication();
+
+  const connect = client.connect({ host, port }, connection, auth);
+
+  logger('Connect client');
+  return { ...state, connect };
```

Comment:
Let's write `connect` to a var, rather than adding it to state. I think we agreed a while back that writing secret stuff to state was probably a bad idea?

Reply:
Also, more importantly, we have to wait for the `client.connect` promise to resolve before we can carry on executing. So you have to do 

```
const connect = await client.connect({ host, port }, connection, auth);
 ```

Reply:
put client as a global variable